### ISSUE TITLE
Add Receipt functionality

### ISSUE THREAD

**Issue by Dextroz** [muyiwaolu/monzo-python#27]
Very recently Monzo teamed up with Flux to bring [paperless receipts](https://monzo.com/blog/2019/01/31/flux-monzo-launch/) to the Monzo App.

I will look into implementing this functionality into the wrapper if others are happy?

Specifically:
- [Create a receipt](https://docs.monzo.com/#create-receipt)
- [Retrieve a receipt](https://docs.monzo.com/#retrieve-receipt) 
- [Delete a receipt](https://docs.monzo.com/#delete-receipt)

**Comment by TomAFrench** [comment]
Unfortunately, it looks like monzo-python would only be able to read receipts which it has created itself.

I'm kinda struggling to see many use cases for this endpoint with this restriction in place. I'd say its worth implementing for completeness but if we can't see retailer-made receipts I can't see what a user would put in a receipt which can't be already added as metadata or notes.

**Comment by Dextroz** [comment]
> Unfortunately, it looks like monzo-python would only be able to read receipts which it has created itself.
> 
> I'm kinda struggling to see many use cases for this endpoint with this restriction in place. I'd say its worth implementing for completeness but if we can't see retailer-made receipts I can't see what a user would put in a receipt which can't be already added as metadata or notes.


@TomAFrench Yes exactly, currently at this time I believe there are only 4 merchants which are creating receipts. The endpoint is mostly for creating your own receipts. As you have pointed out, it does seem kind of pointless as this type of data could just be appended as metadata or notes. 

I made a quick mockup of what possible functionality might look like.

See last 3 functions at the end of [monzo.py](https://github.com/Dextroz/monzo-python/blob/receipt_functionality/monzo/monzo.py)

**Comment by TomAFrench** [comment]
Looks pretty nice, If you make this into a pull request I can point out a couple of changes which need to be made (eg. currently the user won't ever know the `external_id`).

It's worth keeping in mind that the request needs a different encoding than normal. I'm not sure those requests you're making will work but it should just be a small change to account for this.

**Comment by Dextroz** [comment]
Afternoon @TomAFrench, here is the pull request #28.
